lib/trip_dispatcher.rb

```diff
@@ -27,12 +34,41 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def request_trip(passenger_id)
+      first_available_driver = @drivers.find { |driver| driver.status == :AVAILABLE }
+      max_id = @trips.map(&:id).max
+
+      if first_available_driver
+        trip = Trip.new(
+          id: max_id + 1,
+          driver_id: first_available_driver.id,
+          passenger_id: passenger_id,
+          start_time: Time.now,
+          end_time: nil,
+          cost: nil,
+          rating: nil
+        )
+
+        passenger = find_passenger(passenger_id)
+        first_available_driver.start_trip(trip)
+        passenger.add_trip(trip)
+
+        @trips << trip
+
+        return trip
+
+      else
+        raise ArgumentError, "There are no available drivers. Try requesting a trip later."
```

Comment:
This shouldn't be an `ArgumentError` since nothing is wrong with the argument itself.

Generally I'd recommend making a custom error here something like `NoDriversAvailableError`.

```suggestion
        raise ArgumentError, "There are no available drivers. Try requesting a trip later."
```